libs/isr/server/src/utils/cache-utils.spec.ts

```diff
@@ -0,0 +1,38 @@
+import { RenderVariant } from '../../../models/src';
+import { getCacheKey } from './cache-utils';
+
+describe('getCacheKey', () => {
+  it('should return the URL without query parameters when none are allowed', () => {
+    const url = '/page?param1=value1&param2=value2';
+    const result = getCacheKey(url, [], null);
+    expect(result).toBe('/page');
+  });
+
+  it('should return the URL with query parameters when it is null or undefined', () => {
+    const url = '/page?param1=value1&param2=value2';
+    const result = getCacheKey(url, null, null);
+    expect(result).toBe('/page?param1=value1&param2=value2');
+  });
+
+  it('should include only allowed query parameters in the result', () => {
+    const url = '/page?allowed=value&disallowed=value';
```

Comment:
since penetration tests would still try weird things, what would be with the cases:

* `const url = '/page?allowed=value&disallowed=value&allowed=value2';
* `const url = '/page?allowed=value&disallowed=value&allowed=value';

Reply:
btw @maxisam thx for the great work! 🙏

Reply:
ummm, the current behavior is simply just remove the disallowed part and keep all allowed part

so they will be 

- `const url = '/page?allowed=value&allowed=value2';
- `const url = '/page?allowed=value&allowed=value';

I guess we can override the duplicated one ?

so we should make it like

- `const url = '/page?allowed=value2';
- `const url = '/page?allowed=value';

another thought is I actually want to make the getCacheKey like a pluggable function. 

so everyone can have their own cacheKey. 

@eneajaho how do you think about pluggable cacheKey function?

Reply:
@maxisam Yeah we can make the cacheKey function pluggable, good idea.

Reply:
Let's add it in another PR. Just merged this one.